druntime/src/include/importc_msvc_builtins.h

```diff
@@ -0,0 +1,31 @@
+/* This file contains reimplementations of some of the intrinsics recognized
+   by the MSVC compiler, for ImportC.
+   To use it, put `#include <importc_msvc_builtins.h>` in the C source-code that intends to use it.
+
+   This header emits its declarations only when `__IMPORTC_MSVC_BUILTINS__` is not defined.
+   This header defines `__IMPORTC_MSVC_BUILTINS__` when it is included.
+
+   Copyright: Copyright D Language Foundation 2024-2024
+   License: $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+   Authors: Harry Gillanders
+   Source: $(DRUNTIMESRC importc_msvc_builtins.h) */
+
+#ifndef __IMPORTC_MSVC_BUILTINS__
+#define __IMPORTC_MSVC_BUILTINS__ 1
+
+__import __builtins_msvc;
+
+#if defined(__clang__)
+#define __assume(expression) __builtin_assume(expression)
+#elif defined(__GNUC__)
+#define __assume(expression) do {if (!(expression)) {__builtin_unreachable();}} while (0)
+#else
```

Comment:
In case this is meant to be public, FYI I think importC would discard this macro, whereas the other definitions (clang) are simple enough to be exposed as templates.

Reply:
Also `__assume` is supposed to be an optimiser hint, not an assert, which is what this implementation does.

Reply:
(unreachable is a compile-time barrier, not an assert)

I'll take your word for the clang and !clang!gnu paths.

Reply:
> (unreachable is a compile-time barrier, not an assert)

I know that. `assert` is (often) defined as `do { if (!cond) { /* print stuff*/ ; __builtin_unreachable();}} while (0)`. This is almost identical (modulo the print stuff).

Assume should be `#define assume(x) x`, i.e. a no-op.